luarules/gadgets/cmd_nano_range_check.lua

```diff
@@ -0,0 +1,160 @@
+function gadget:GetInfo()
+	return {
+		name = "Construction Turrets Range Check",
+		desc = "Stops construction turrets from getting stuck on orders out of reach.",
+		author = "Nehroz",
+		date = "2024.12.01",
+		layer = 0,
+		enabled = true,
+	}
+end
+
+if not gadgetHandler:IsSyncedCode() then return end
+
+local CMD_STOP = CMD.STOP
+local gMaxUnits = Game.maxUnits
+local updateInterval = Game.gameSpeed / 2
+
+local spValidUnitID = Spring.ValidUnitID
+local spValidFeatureID = Spring.ValidFeatureID
+local spGetUnitDefID = Spring.GetUnitDefID
+local spGetUnitSeparation = Spring.GetUnitSeparation
+local spGetUnitFeatureSeparation = Spring.GetUnitFeatureSeparation
+local spGetUnitCommands = Spring.GetUnitCommands
+local spGiveOrderToUnit = Spring.GiveOrderToUnit
+
+--- Key is the nano's ID, followed by a set-like with the unitID of target currently inside the nano's range.
+--- Used for CommandArray manipulation
+---@type {[integer] : {integer : {cmdTag : integer, status : integer}}}
+local trackingTable = {}
+-- Limits how many GetUnitCommands are called inside a frame when required.
+-- higher values allow faster tagging synchronization after a new command;
+-- which can cause proportional lag when many units are targeted. (long tables returned)
+local tagUpdateFrameMax = 1
```

Comment:
since I think `tagUpdateFrameMax` here is a constant maybe should use all caps, but no prob if you rather like this, just sayin

Reply:
I was tolled not to use all caps because it's Lua. I prefer to all cap for non mut vars as well.